### DISCUSSION TITLE
The best approach for building an admin panel with Nextjs

### DISCUSSION THREAD

**Discussion by tareefdev** [vercel/next.js#18901] [Help]
I'm going to build an admin panel using Nextjs which is capable of listing, creating, and editing data from a database (that has one million media item), but I'm considering which approach achieves better performance in my case:

a) Using static generation pages with a client-side rendering for data from the database (`swr`)
or
b) Using server-side rendering pages. (`getServerSideProps`)

As I understand: (a) means more TTFB than the second option, but it also has better performance as a better CPU is going to render the page rather than the client's CPU which could be overloaded by other pages and other apps. On the other hand, other discussions here mentioned that the (a) option has better caching, which results in a better browsing experience as the same users are going to use the same database over and over again.

Are those the only aspects of comparison in this case?

The app is private only a few users are going to use it. Obviously SEO is not a concern.  I'm not sure if I forget some other factors that are involving here too.

If my understanding is accurate, what would result in a better performant site: a server-side rendering (taking the heavy load from the client) or a better-caching policy?

**Comment by watch-janick** [comment]
@tareefdev I would definitely go for static generation + SWR.

I've been using it to build an admin panel recently and the performances are just awesome. I wouldn't be too worried about the CPU performances of the client, in today's world there little chances for this to be an issue.

**Comment by isanjayjoshi** [comment]
This video guides you through building a powerful, modern admin dashboard using Next.js - https://www.youtube.com/watch?v=PjjJLKZp9xI

![image](https://github.com/user-attachments/assets/fc665081-0654-4551-864d-646ff4bd8e3c)
